app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt

```diff
@@ -53,6 +54,10 @@ class MainActivity : SimpleActivity() {
 
         flashlight_btn.setOnClickListener {
             mCameraImpl!!.toggleFlashlight()
+            if (mCameraImpl?.supportsBrightnessControl() == true) {
+                brightness_bar.beInvisibleIf(brightness_bar.isVisible)
```

Comment:
lets not just toggle the bar visibility here blindly, rely on some flashlight status or so to make it more reliable. Currently if Im toggling flashlight and morse + seekbars, it can happen that the brightness control works wrong, attaching a screenshot. It works in a flipped way that the top flashlight icon uses the primary color properly, but the brightness bar is shown only when the flashlight is turned off.

![flipped](https://user-images.githubusercontent.com/528931/198045731-8a68b682-0eb7-4552-af5a-ea10db639a46.png)

Reply:
Added this. Now the seekbar would be shown based on a listener from the Camera API.